packages/addons/tools/jre.zulu/package.mk

```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2019-present Peter Vicman (peter.vicman@gmail.com)
+
+PKG_NAME="jre.zulu"
+PKG_VERSION="1.0"
+PKG_REV="100"
+PKG_LICENSE="GPL2"
+PKG_DEPENDS_TARGET="jdk-${TARGET_ARCH}-zulu jre-libbluray libXext chrome-libXtst chrome-libXi chrome-libXrender jre-libXinerama"
+PKG_SECTION="tools"
+PKG_SHORTDESC="Java Runtime Environment 8 for Blu-ray Disc Java menus from Azul Systems."
+PKG_LONGDESC="$PKG_SHORTDESC"
+PKG_TOOLCHAIN="manual"
+
+PKG_IS_ADDON="yes"
+PKG_ADDON_NAME="JRE for BD-J menus"
+PKG_ADDON_TYPE="xbmc.python.script"
+
+# find $1.so.[0-9]* in $2 and copy it to dest
+_pkg_copy_lib() {
+  find "$2" -regextype sed -regex ".*/$1\.so\.[0-9]*" \
+    -exec cp {} "$ADDON_BUILD/$PKG_ADDON_ID/lib" \;
```

Comment:
This is only copying the major version, and not the other major/minor and unversioned libraries.

For example, given the following:
```
ubuntu@buildbox:~/projects/LibreELEC.tv-test/build.LibreELEC-RPi2.arm-9.80-devel/chrome-libXtst-1.2.3$ ls -la .armv7ve-libreelec-linux-gnueabi/src/.libs
total 56
drwxrwxr-x 2 ubuntu ubuntu  4096 Jul  7 02:48 .
drwxrwxr-x 4 ubuntu ubuntu  4096 Jul  7 02:48 ..
lrwxrwxrwx 1 ubuntu ubuntu    13 Jul  7 02:48 libXtst.la -> ../libXtst.la
-rw-rw-r-- 1 ubuntu ubuntu  1500 Jul  7 02:48 libXtst.lai
lrwxrwxrwx 1 ubuntu ubuntu    16 Jul  7 02:48 libXtst.so -> libXtst.so.6.1.0
lrwxrwxrwx 1 ubuntu ubuntu    16 Jul  7 02:48 libXtst.so.6 -> libXtst.so.6.1.0
-rwxrwxr-x 1 ubuntu ubuntu 24248 Jul  7 02:48 libXtst.so.6.1.0
-rw-rw-r-- 1 ubuntu ubuntu 11608 Jul  7 02:48 XRecord.o
-rw-rw-r-- 1 ubuntu ubuntu  6792 Jul  7 02:48 XTest.o
```
then only the following appears in the add-on:
```
ubuntu@buildbox:~/projects/LibreELEC.tv-test/build.LibreELEC-RPi2.arm-9.80-devel/addons/jre.zulu/tools.jre.zulu$ ls -la lib
total 1280
drwxrwxr-x 2 ubuntu ubuntu    4096 Jul  7 02:48 .
drwxrwxr-x 6 ubuntu ubuntu    4096 Jul  7 02:48 ..
-rwxrwxr-x 1 ubuntu ubuntu 1124652 Jul  7 02:48 libX11.so.6
-rwxrwxr-x 1 ubuntu ubuntu   55368 Jul  7 02:48 libXext.so.6
-rwxrwxr-x 1 ubuntu ubuntu    9516 Jul  7 02:48 libXinerama.so.1
-rwxrwxr-x 1 ubuntu ubuntu   50608 Jul  7 02:48 libXi.so.6
-rwxrwxr-x 1 ubuntu ubuntu   30248 Jul  7 02:48 libXrender.so.1
-rwxrwxr-x 1 ubuntu ubuntu   17816 Jul  7 02:48 libXtst.so.6
```
ie. the "major" sym link version has been dereferenced and copied, but not the others.

I assume this isn't an issue, but is there a reason to only copy the major version (and not the major/minor version, or the unversioned sym link?)

If there's no particular reason to only copy the major version, then the following standard `cp` could be used instead of adding the extra function with `find .. -exec cp` etc.:
```
cp $(get_build_dir chrome-libXtst)/.$TARGET_NAME/src/.libs/libXtst.so* "$ADDON_BUILD/$PKG_ADDON_ID/lib"
```

This [patch](http://ix.io/1NTK) is working fine, producing the following library files in the finished add-on:
```
ubuntu@buildbox:~/projects/LibreELEC.tv-test/build.LibreELEC-RPi2.arm-9.80-devel/addons/jre.zulu/tools.jre.zulu$ ls -la lib
total 1280
drwxrwxr-x 2 ubuntu ubuntu    4096 Jul  7 03:44 .
drwxrwxr-x 6 ubuntu ubuntu    4096 Jul  7 03:44 ..
lrwxrwxrwx 1 ubuntu ubuntu      15 Jul  7 03:44 libX11.so -> libX11.so.6.3.0
lrwxrwxrwx 1 ubuntu ubuntu      15 Jul  7 03:44 libX11.so.6 -> libX11.so.6.3.0
-rwxrwxr-x 1 ubuntu ubuntu 1124652 Jul  7 03:44 libX11.so.6.3.0
lrwxrwxrwx 1 ubuntu ubuntu      16 Jul  7 03:44 libXext.so -> libXext.so.6.4.0
lrwxrwxrwx 1 ubuntu ubuntu      16 Jul  7 03:44 libXext.so.6 -> libXext.so.6.4.0
-rwxrwxr-x 1 ubuntu ubuntu   55368 Jul  7 03:44 libXext.so.6.4.0
lrwxrwxrwx 1 ubuntu ubuntu      20 Jul  7 03:44 libXinerama.so -> libXinerama.so.1.0.0
lrwxrwxrwx 1 ubuntu ubuntu      20 Jul  7 03:44 libXinerama.so.1 -> libXinerama.so.1.0.0
-rwxrwxr-x 1 ubuntu ubuntu    9516 Jul  7 03:44 libXinerama.so.1.0.0
lrwxrwxrwx 1 ubuntu ubuntu      14 Jul  7 03:44 libXi.so -> libXi.so.6.1.0
lrwxrwxrwx 1 ubuntu ubuntu      14 Jul  7 03:44 libXi.so.6 -> libXi.so.6.1.0
-rwxrwxr-x 1 ubuntu ubuntu   50608 Jul  7 03:44 libXi.so.6.1.0
lrwxrwxrwx 1 ubuntu ubuntu      19 Jul  7 03:44 libXrender.so -> libXrender.so.1.3.0
lrwxrwxrwx 1 ubuntu ubuntu      19 Jul  7 03:44 libXrender.so.1 -> libXrender.so.1.3.0
-rwxrwxr-x 1 ubuntu ubuntu   30248 Jul  7 03:44 libXrender.so.1.3.0
lrwxrwxrwx 1 ubuntu ubuntu      16 Jul  7 03:44 libXtst.so -> libXtst.so.6.1.0
lrwxrwxrwx 1 ubuntu ubuntu      16 Jul  7 03:44 libXtst.so.6 -> libXtst.so.6.1.0
-rwxrwxr-x 1 ubuntu ubuntu   17816 Jul  7 03:44 libXtst.so.6.1.0
```

Reply:
Ah, I see - the `zip` archive created by `7za` isn't handling the sym links - it dereferences the sym link and copies the same library file 3 times. Eugh. Apparently `7za` has sym link support, I'll take a look at that...

Reply:
Bah, `7za` sym link support only extends as far as WIM (Windows Image Format) and TAR files, not zip. OK, now I understand what you are doing - strange that I'd not seen this space-saving measure before, or maybe I've just completely overlooked it...

This does in fact seem to be a limitation of `7za`, as the standard `zip` compression tool *DOES* support symbolic links (and is also significantly quicker than `7za` without being obviously less efficient).

Unfortunately, Kodi doesn't support the symbolic links when unzipping the archive so we're back to square one. Bummer. OK ignore this second patch... sorry for the rambling... :)

Reply:
No point of copying unneeded files :-)